cargo-dist/src/build/generic.rs

```diff
@@ -168,31 +199,30 @@ pub fn build_generic_target(
 
 /// Similar to the above, but with slightly different signatures since
 /// it's not based around axoproject-identified binaries
-pub fn run_extra_artifacts_build(
-    dist_graph: &DistGraph,
-    target: &ExtraBuildStep,
-) -> DistResult<()> {
+pub fn run_extra_artifacts_build(dist: &DistGraph, build: &ExtraBuildStep) -> DistResult<()> {
     eprintln!(
         "building extra artifacts target (via {})",
-        target.build_command.join(" ")
+        build.build_command.join(" ")
     );
 
-    let result = run_build(dist_graph, &target.build_command, None)?;
-    let dest = dist_graph.dist_dir.to_owned();
+    // TODO: run this in build.working_dir (conflicts with other inflight PR)
```

Comment:
this is referring to #1116 which changes the same code and similarly wants to make commands run in a working dir

Reply:
Added a second commit that adds duped work from the PR, as the tests needed it to pass.